EXSWHTEC-281 - Implement Unit Tests for assert functions

milos-mozetic opened the pull request:
Positive and negative tests are implemented for:
- assert, i.e. run-time assert
- static_assert, i.e. compile-time assert

milos-mozetic committed:
EXSWHTEC-281 - Implement basic unit tests for assert and static_assert functions

milos-mozetic committed:
EXSWHTEC-281 - Adjust negative kernel names

milos-mozetic committed:
EXSWHTEC-281 - Add messages to the static_assert calls to resolve compiler warnings

milos-mozetic committed:
EXSWHTEC-281 - Add doxygen comments

milos-mozetic committed:
EXSWHTEC-281 - Refactor doxygen comments

milos-mozetic committed:
EXSWHTEC-281 - Rename Assert and Static Assert groups to be equal to the API definition

milos-mozetic committed:
EXSWHTEC-281 - Fix year in the Copyright comment

milos-mozetic committed:
EXSWHTEC-281 - Apply clang-format

milos-mozetic committed:
EXSWHTEC-281 - Adjust tests to work on NVIDIA

As it can be observed, static_assert is not available within NVRTC, and hipDeviceSynchronize on NVIDIA returns hipErrorAssert instead of abort signal.

milos-mozetic committed:
EXSWHTEC-281 - Remove REQUIRE for abort_raised_flag in case of NVIDIA platform

milos-mozetic committed:
EXSWHTEC-281 - Adjust comments for NVIDIA case of assert()

milos-mozetic committed:
EXSWHTEC-281 - Remove unnecessary set(TEST_SRC) from CMakeLists.txt

chrispaquot commented on catch/unit/assertion/assert.cc:
```diff
@@ -0,0 +1,118 @@
+
+  if constexpr (should_abort) {
+    AssertFailKernel<<<num_blocks, num_threads, 0, 0>>>();
+#if HT_AMD
```
add the test to json to skip it on AMD.

MMozetic replied:
@chrispaquot: Can you please describe what is the issue here? Additionally, is it causing issues on Windows or Linux?

milos-mozetic replied:
@chrispaquot: Can you please describe what is the issue here? Additionally, is it problematic on Windows or Linux?

milos-mozetic committed:
EXSWHTEC-281 - Refactor compileAndCaptureOutput.py

rakesroy committed:
Merge branch 'develop' into assert_device_tests

milos-mozetic committed:
EXSWHTEC-281 - Remove mistakenly included hip_test_defgroups.hh from source files and add guard to the hip_test_defgroups.hh to prevent this kind of errors

The header file hip_test_defgroups.hh is intended to be used only for doxygen groups definitions, as well as test case stubs when the test case is generated with a macro or some other case
that cannot be identified by doxygen. This file doesn't contain any usable code, but only contains comments and stub test cases. Therefore, it shall not be included anywhere.
Doxygen is able to detect groups from this header file, even if it is not included anywhere because doxygen is not performing build, rather is reading information from text files.

milos-mozetic commented:
@rakesroy, please note that the issue with this PR is resolved within commit: https://github.com/ROCm-Developer-Tools/hip-tests/pull/210/commits/557446b622e447dcf1e6e4b278d959fcc2bef7d1. Can you please check again that the problem is solved now?

rakesroy commented:
@milos-mozetic ,
Due to recent pushes to develop branch, merge conflicts occurred in this PR.
Will you please resolve them?

milos-mozetic committed:
Merge branch 'develop' of github.com:mirza-halilcevic/hip-tests into assert_device_tests

milos-mozetic commented:
@rakesroy: Merge conflict has been resolved within commit https://github.com/ROCm-Developer-Tools/hip-tests/pull/210/commits/73fdae115f219231ea88e28fdee30812f455a929.

rakesroy committed:
Update config_amd_linux_common.json

- Disable tests which fail in external CI

rakesroy committed:
Update config_nvidia_linux_common.json

rakesroy committed:
Update config_amd_windows_common.json

- Disable tests which fail in external CI

rakesroy committed:
Merge branch 'develop' into assert_device_tests

mirza-halilcevic committed:
Merge remote-tracking branch 'origin/develop' into assert_device_tests

rakesroy commented:
PR has been merged into develop branch via commit 044a59496cbfec069485d9233d93d96f46927565.
